### PR TITLE
Added Logging Configuration

### PR DESCRIPTION
**Pull request by adomas-vensas**
- Added `IOptionsMonitor` to change LoggingMiddleware options at runtime.
- Added a few logging options
- Made so that multiple roles are logged

Had some great input and help from @TomasKvaraciejus 

Closes issue: #49

### CODE REVIEW

File: Recipes/Recipes.API/Middlewares/LoggingMiddleware.cs [hunk 1]
```diff
@@ -8,16 +9,30 @@ public class LoggingMiddleware
...
     public LoggingMiddleware(RequestDelegate next,
-    ILogger<LoggingMiddleware> logger)
+    ILogger<LoggingMiddleware> logger,
+    IOptionsMonitor<LoggerOptions> options
+    )
     {
         _next = next;
         _logger = logger;
+
+        options.OnChange(option =>
```
Comment (6ar8nas) [inline Recipes/Recipes.API/Middlewares/LoggingMiddleware.cs:22]: Do you save the initial options though?

File: Recipes/Recipes.API/Middlewares/LoggingMiddleware.cs [hunk 2]
```diff
@@ -8,16 +9,30 @@ public class LoggingMiddleware
 {
     private readonly RequestDelegate _next;
     private readonly ILogger<LoggingMiddleware> _logger;
+    private LoggerOptions _options;
 
     public LoggingMiddleware(RequestDelegate next,
```
Comment (Rytis-Sapka) [inline Recipes/Recipes.API/Middlewares/LoggingMiddleware.cs:14]: This part seems a bit sus, github actions is also giving errors. Maybe to be safe we should set _options on startup

Comment (Rytis-Sapka) [reply Recipes/Recipes.API/Middlewares/LoggingMiddleware.cs:14]: image

File: Recipes/Recipes.API/Program.cs [hunk 3]
```diff
@@ -45,10 +45,14 @@
...
 builder.Services.AddScoped(typeof(IGenericRepository<>), typeof(GenericRepository<>));
 builder.Services.AddScoped<IRecipesRepository, RecipesRepository>();
+builder.Services.AddSingleton<LoggerOptions, LoggerOptions>();
 
 builder.Services.ConfigureOptions<ConfigureJwtBearerOptions>();
 builder.Services.Configure<JwtOptions>(
     builder.Configuration.GetRequiredSection(JwtOptions.SectionName));
+builder.Services.Configure<LoggerOptions>(
+    builder.Configuration.GetRequiredSection(LoggerOptions.SectionName)
+    );
```
Comment (Rytis-Sapka) [inline Recipes/Recipes.API/Program.cs:55]: Invonsistent formatting with line 52

File: Recipes/Recipes.API/Middlewares/LoggingMiddleware.cs (hunk 2 above)
Comment (adomas-vensas) [reply Recipes/Recipes.API/Middlewares/LoggingMiddleware.cs:14]: Yes, already fixed it

File: Recipes/Recipes.API/Program.cs (hunk 3 above)
Comment (adomas-vensas) [reply Recipes/Recipes.API/Program.cs]: Did it

File: Recipes/Recipes.API/Middlewares/LoggingMiddleware.cs (hunk 1 above)
Comment (adomas-vensas) [reply Recipes/Recipes.API/Middlewares/LoggingMiddleware.cs:23]: Yup, fixed it

**Review by TomasKvaraciejus — APPROVED** [review]
very cool :)

**Pull request merged**